crates/cheatcodes/src/evm.rs

```diff
@@ -1044,3 +1096,49 @@ fn genesis_account(account: &Account) -> GenesisAccount {
         private_key: None,
     }
 }
+
+/// Helper function to returns state diffs recorded for each changed account.
+fn get_recorded_state_diffs(state: &mut Cheatcodes) -> StateDiffs {
+    let mut state_diffs: StateDiffs = BTreeMap::default();
+    if let Some(records) = &state.recorded_account_diffs_stack {
+        let mut recorded_changes: BTreeMap<Address, BTreeMap<U256, Vec<(B256, B256)>>> =
+            BTreeMap::default();
+        records
+            .iter()
+            .flatten()
+            .filter(|account_access| !account_access.storageAccesses.is_empty())
+            .for_each(|account_access| {
+                for storage_access in &account_access.storageAccesses {
+                    // Retain only records with storage accesses writes that didn't revert.
+                    if storage_access.isWrite && !storage_access.reverted {
+                        recorded_changes
+                            .entry(storage_access.account)
+                            .or_default()
+                            .entry(storage_access.slot.into())
+                            .or_default()
+                            .push((storage_access.previousValue, storage_access.newValue));
+                    }
+                }
+            });
+
+        for (address, recorded_changes) in recorded_changes {
+            let mut changes = BTreeMap::default();
+            for (slot, recorded_slot_changes) in recorded_changes {
+                // For each slot we retain the initial value from first state change and the new
+                // value from last state change.
+                changes.insert(
+                    slot.to_string(),
+                    SlotStateDiff {
+                        original: recorded_slot_changes.first().unwrap().0,
+                        dirty: recorded_slot_changes.last().unwrap().1,
```

Comment:
I think we should consider sticking to `previousValue` / `newValue` here as, in the current PR, we are not proposing to change the fields of the `StorageAccess` struct

Alternatively we could update the struct with the `original`, `dirty` naming

Reply:
updated in https://github.com/foundry-rs/foundry/pull/9435/commits/4e684f8eca67ad28e3c7d85e53706581907f189f